src/JokeRandomiser.tsx

```diff
@@ -1,14 +1,77 @@
 import styled from "styled-components";
 import BackButton from "./BackButton";
 import NextButton from "./NextButton";
-import React from "react";
+import React, { useCallback, useState } from "react";
+import { Button } from "../stories/Button";
+
+// Define a type for the Joke object
+interface Joke {
+  id: number;
+  type: string;
+  setup: string;
+  punchline: string;
+}
 
 export default function JokeRandomiser() {
+  // Render json from fetchJoke() when handleClick() event fires
+  const [joke, setJoke] = useState<Joke | null>(null);
+  //  useState with boolean type to initially hide the punchline by default
+  const [isHidden, setIsHidden] = useState<boolean>(false);
+  // useCallback to toggle the boolean state of the punchline
+  const toggleJokePunchline = useCallback(
+    () => setIsHidden((isHidden) => !isHidden),
+    [setIsHidden]
+  );
+
+  // Declare an async function to fetch a random joke
+  const randomJoke = async (): Promise<Joke> => {
```

Comment:
I love this new api you have found, this has made our app so awesome, I love it, amazing work ❤️